aeron-client/src/main/c/aeron_subscription.c

```diff
@@ -527,8 +530,100 @@ int64_t aeron_header_position(aeron_header_t *header)
         header->frame->term_id, offset_at_end_of_frame, header->position_bits_to_shift, header->initial_term_id);
 }
 
+int aeron_subscription_local_sockaddrs(
+    aeron_subscription_t *subscription, aeron_iovec_t *address_vec, size_t address_vec_len)
+{
+    if (NULL == subscription || address_vec == NULL || address_vec_len < 1)
+    {
+        errno = EINVAL;
+        aeron_set_err(EINVAL, "%s", strerror(EINVAL));
+        return -1;
+    }
+
+    return aeron_local_sockaddr_find_addrs(
+        &subscription->conductor->counters_reader,
+        subscription->channel_status_indicator_id,
+        address_vec,
+        address_vec_len);
+}
+
+int aeron_subscription_resolved_endpoint(
+    aeron_subscription_t *subscription, char *address, size_t address_len)
+{
+    if (NULL == subscription || address == NULL || address_len < 1)
+    {
+        errno = EINVAL;
+        aeron_set_err(EINVAL, "%s", strerror(EINVAL));
+        return -1;
+    }
+
+    aeron_iovec_t addr_vec;
+    addr_vec.iov_base = (uint8_t *)address;
+    addr_vec.iov_len = address_len;
+
+    return aeron_local_sockaddr_find_addrs(
+        &subscription->conductor->counters_reader,
+        subscription->channel_status_indicator_id,
+        &addr_vec,
+        1);
+}
+
+int aeron_subscription_try_resolve_channel_endpoint_port(
+    aeron_subscription_t *subscription,
+    char *address,
+    size_t address_len)
+{
+    if (NULL == subscription || address == NULL || address_len < 1)
+    {
+        errno = EINVAL;
+        aeron_set_err(EINVAL, "%s", strerror(EINVAL));
+        return -1;
+    }
+
+    aeron_uri_t temp_uri;
+    char resolved_endpoint[AERON_CLIENT_MAX_LOCAL_ADDRESS_STR_LEN];
+    memset(&temp_uri, 0, sizeof(temp_uri));
```

Comment:
Better to initialize with `aeron_uri_t temp_uri = { 0 }`

Reply:
I did not know you could do that.

Reply:
C yes. For C++ you need to put nothing in the braces. Shame they are not consistent.